features/old/api/features.feature

```diff
@@ -8,59 +8,60 @@ Feature: Features :)
     And an application plan "Basic" of provider "foo.3scale.localhost"
     Given current domain is the admin domain of provider "foo.3scale.localhost"
 
-  @javascript @wip
+  @javascript
   Scenario: Create a feature
+    Given a service plan "Basic" of provider "foo.3scale.localhost"
     When I log in as provider "foo.3scale.localhost"
-    And I go to the edit page for plan "Basic"
+    Then I go to the edit page for plan "Basic"
     And I follow "New feature"
     And I fill in "Name" with "Free T-shirt"
+    And I fill in "System name" with "tee"
     And I fill in "Description" with "T-shirt with logo of our company for free."
-    And I press "Create Feature"
-    Then I should see enabled feature "Free T-shirt"
-    And provider "foo.3scale.localhost" should have feature "Free T-shirt"
-    And feature "Free T-shirt" should be enabled for plan "Basic"
+    And I press "Save"
+    Then I should see "T-shirt with logo of our company for free."
 
   # TODO: Test that the "No features yet" notice dissapears when first feature is created,
   #       and appears when last one is deleted.
-
-  @javascript @wip
+  @javascript
   Scenario: Disable a feature
-    Given a feature "50% less bugs" of provider "foo.3scale.localhost"
-    And feature "50% less bugs" is enabled for plan "Basic"
+    Given a service plan "Basic" of provider "foo.3scale.localhost"
+    And a feature "50% more bugs" of provider "foo.3scale.localhost"
+
     When I log in as provider "foo.3scale.localhost"
-    And I go to the edit page for plan "Basic"
-    And I press the disable button for feature "50% less bugs"
-    Then I should see disabled feature "50% less bugs"
-    And feature "50% less bugs" should be disabled for plan "Basic"
+    Then I go to the edit page for plan "Basic"
+    And I disable feature "50% more bugs"
+    Then feature "50% more bugs" should be disabled
 
-  @javascript @wip
+  @javascript
   Scenario: Enable a feature
-    Given a feature "Light-speed connection" of provider "foo.3scale.localhost"
-    And feature "Light-speed connection" is disabled for plan "Basic"
+    Given a service plan "Basic" of provider "foo.3scale.localhost"
+    And a feature "50% less bugs" of provider "foo.3scale.localhost"
+
     When I log in as provider "foo.3scale.localhost"
-    And I go to the edit page for plan "Basic"
-    And I press the enable button for feature "Light-speed connection"
-    Then I should see enabled feature "Light-speed connection"
-    And feature "Light-speed connection" should be enabled for plan "Basic"
+    Then I go to the edit page for plan "Basic"
+    And I enable feature "50% less bugs"
```

Comment:
These steps "I enable feature", "should be enabled" etc. only test the logic on the model level, they have nothing to do with the UI (which is the main purpose of the cucumber tests), and as I mentioned earlier these steps for some reason are for the service plans, while in the UI the page that is loaded is the application plan.

The steps should interact with the UI elements - in case of enabling/disabling the feature it should be clicking on the icon in the **Enabled?** column of the features table and verifying that the icon has changed (and additionally we can check if on the DB level the attribute has changed accordingly - to prove that it's not only a visual thing).

Reply:
Okay  let me recheck this